lldb/source/Core/ValueObjectChild.cpp

```diff
@@ -168,19 +169,25 @@ bool ValueObjectChild::UpdateValue() {
             else
               m_value.SetValueType(Value::eValueTypeFileAddress);
           } break;
-          case eAddressTypeLoad:
+          case eAddressTypeLoad: {
             // BEGIN SWIFT MOD
             // We need to detect when we cross TypeSystem boundaries,
             // e.g. when we try to print Obj-C fields of a Swift object.
-            if (parent->GetCompilerType().GetTypeSystem()->getKind() ==
-                GetCompilerType().GetTypeSystem()->getKind())
+            bool same_type_system = llvm::isa<ClangASTContext>(parent->GetCompilerType().GetTypeSystem()) &&
```

Comment:
When did this change/break?

Reply:
When I removed the Kind enum from TypeSystem (see the commit for the change).

Reply:
What was the rationale for that? The new way of writing this code looks clumsy and error prone.

Reply:
That we don't have a centralized enum that has to list all TypeSystems inside LLDB (which would need to list all Mock and testing TypeSystems).

Anyway, the actual change that broke this is that the new RTTI implementation details are private (the ID member) so we can't just hijack that for this check. I could also make the RTTI backend public again but that would be a downstream patch. Or I upstream some kind of check that allows to check if two TypeSystems are of the same class.